### ISSUE TITLE
Missing Accept Header

### ISSUE THREAD

**Issue by gbonnefille** [mattermost/mattermost-mobile#4760]
Submit feature requests to http://www.mattermost.org/feature-requests/. File non-security related bugs here in the following format: 

#### Summary
It seems that Accept Header is missing.

#### Environment Information

- Device Name: Android
- OS Version: 7
- Mattermost App Version: 1.34.1
- Mattermost Server Version:

#### Steps to reproduce
Normal login.

#### Expected behavior
Our ModSecurity complaints about missing such headers.

#### Observed behavior
Hard to elaborate as I observed this warning while investigating on #4527 

#### Possible fixes
Not sure (not JS developer), but would certainly be simple to enforce such headers in requests.

**Comment by enahum** [comment]
@gbonnefille Thanks for the report. After having an internal convo with the team I got the following:

The Mattermost Server does not require the Accept headers and has suitable default content types set when they are not present. In fact none of our clients set it.

I s this a real security issue? Maybe not, maybe is just that "best practice" scanning is likely to pick it up in a generic way.

**Comment by larkox** [comment]
Closing due to the age of the ticket. If this becomes an issue, feel free to reopen the ticket.

**Comment by AnnoyingTechnology** [comment]
Still an issue, it would be great to reopen

**Comment by larkox** [comment]
@AnnoyingTechnology Sure. Can you explain how is this an issue for you?

I would like to understand the problematic and see if there is any way we can solve it.

**Comment by jupenur** [comment]
920300 is a non-blocking rule -- so yes, it should complain but never block. Not sending Accept headers is entirely in compliance with the HTTP spec. I don't see an issue here, but not opposed to adding Accept headers either.

**Comment by AnnoyingTechnology** [comment]
hi, @amyblais is it fixed though ?

**Comment by enahum** [comment]
@panoramix360 yes, that would apply to every request BUT it may break some, so I guess the best here would be to add to a per-request basis

**Comment by AnnoyingTechnology** [comment]
> Do you think it's worth it to work on this issue?

As a user affected by the issue, I'd say yes, in my own interest :)

**Comment by panoramix360** [comment]
Created this Draft PR so we can test it and discuss.

**Comment by panoramix360** [comment]
The PR with the fix got merged, it's probably going to be available on the next release!

Let me know if ModSecurity works for you this time, thank you!
